app/src/test/kotlin/fr/free/nrw/commons/media/MediaDetailFragmentUnitTests.kt

```diff
@@ -229,6 +231,22 @@ class MediaDetailFragmentUnitTests {
         fragment.launchZoomActivity(view)
     }
 
+    @Test
+    @Throws(Exception::class)
+    fun testOnUpdateCoordinatesClicked() {
+        `when`(media.coordinates).thenReturn(null)
+        `when`(applicationKvStore.getString(LAST_LOCATION)).thenReturn("37.773972,-122.431297")
+        fragment.onUpdateCoordinatesClicked()
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun testOnUpdateCoordinatesClickedNotNullValue() {
+        `when`(media.coordinates).thenReturn(LatLng(-0.000001, -0.999999, 0f))
+        `when`(applicationKvStore.getString(LAST_LOCATION)).thenReturn("37.773972,-122.431297")
+        fragment.onUpdateCoordinatesClicked()
```

Comment:
Here you can verify if `media.getCoordinates()` is invoked only 1 time or not.

You can also Assert for Shadow Activity Intent and check if LocationPickerActivity is launched or not.